addons/website_sale/controllers/main.py

```diff
@@ -534,6 +534,7 @@ def values_postprocess(self, order, mode, values, errors, error_msg):
                     _logger.debug("website_sale postprocess: %s value has been dropped (empty or not writable)" % k)
 
         new_values['customer'] = True
+        new_values['company_id'] = request.website.user_id.company_id.id
```

Comment:
Wouldn't it be `request.website.company_id.id` better?

Reply:
Well, I have chosen this one because maybe you want to leave `company_id` in public user empty for total visibility of the partner in all companies.

Reply:
Oh, right :smile: :+1:

Reply:
You're both right. Changed.